ticker/ticker.go

```diff
@@ -60,10 +60,10 @@ type Ticker interface {
 	Stop()
 }
 
-// ticker is the production implementation of the resumable Ticker interface.
-// This allows various components to toggle their need for tick events, which
-// may vary depending on system load.
-type ticker struct {
+// T is the production implementation of the resumable Ticker interface. This
```

Comment:
The name `T` is a bit confusing. What's the rationale behind the change?

Reply:
`ticker` should have previously been exported, but renaming it to `Ticker` would conflict with the interface, so @cfromknecht and I settled with `T`.

Reply:
Why does it have to be exported? The commit message should have an explanation at least IMO.

Reply:
Fixed.

Reply:
Still don't see why exporting it is needed. Having (just) an exported interface nice since it hides the actual implementation details from the user of the module. Are we actually (planning to be) using the fact that the ticker implementation is now exported?

Reply:
It's just good practice to return exported structs. This doesn't apply to this particular situation, but let's say there's the struct satisfies another interface which you'd like to use. If you return the interface and the struct remains unexported, there's no way of using that same struct to satisfy other interfaces.

> Having (just) an exported interface nice since it hides the actual implementation details from the user of the module.

How does exporting the struct change this? You still interact with it the same way.

Reply:
@halseth it also fixes the godocs https://godoc.org/github.com/lightningnetwork/lnd/ticker